### PR TITLE
Footnote support

### PR DESCRIPTION
**Pull request by paulwarren-wk**
Add support for viewing footnotes in an iXBRL document.

When selecting a fact which has one or more footnotes, the footnotes will be listed in the "footnotes" section of the inspector.  Clicking on a footnote focus on the footnote, and the inspector will show a list of facts with which that footnote is associated.

Footnotes can also be selected directly by clicking in iXBRL view, and will be highlighted by "highlight all tags".

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(1) Security relevant changes were detected</summary>
<li>Watched keyword <code>.html(t</code> in <code>iXBRLViewerPlugin/viewer/src/js/inspector.js</code> line(s) ['494'] added</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by brettkail-wk** [comment]
@paulwarren-wk Build has test failures.

**Comment by paulwarren-wk** [comment]
JS tests now passing, still working on the Python tests.

File: iXBRLViewerPlugin/viewer/src/js/inspector.js [hunk 1]
```diff
@@ -470,55 +485,55 @@ Inspector.prototype.update = function () {
...
+            this._currentFactList.push(this._report.getItemById(factIdList[i]));
         }
     }
-    this.switchFact(id);
+    this.switchItem(id);
 }
 
 /*
- * Switches the currently selected fact.  Unlike selectFact, this does not
+ * Switches the currently selected fact.  Unlike selectItem, this does not
```
Comment (joyreistad-wf) [inline iXBRLViewerPlugin/viewer/src/js/inspector.js]: should this terminology be updated to not be fact specific?

Comment (paulwarren-wk) [reply iXBRLViewerPlugin/viewer/src/js/inspector.js]: Have attempted to clarify in e771d30, although your comment has made me realise that we probably ought to support nested footnotes, as well as facts nested inside footnotes and vice versa.

I'll tackle that separately.

**Comment by paulwarren-wk** [comment]
Have now merged the code for coping with nested footnotes and facts into this PR as @derekgengenbacher-wf  QA has uncovered this as an issue.

File: samples/src/continuation/continuation.html [hunk 2]
```diff
@@ -75,7 +75,10 @@ <h1>ix:continuation test document</h1>
             This is the content of footnote 2.  It's much longer that footnote 1 and has multiple paragraphs.
         </p>
         <p>
-            It also has <b>bold</b> and <i>italic</i> text.
+            <ix:footnote id="fn4">
+            It also has <b>bold</b> and <i>italic</i> text, and contains a nested footnote.
```
Comment (joyreistad-wf) [inline samples/src/continuation/continuation.html]: nit indentation

File: iXBRLViewerPlugin/viewer/src/js/report.js [hunk 3]
```diff
@@ -36,15 +36,22 @@ iXBRLReport.prototype.setIXNodeMap = function(ixData) {
...
+    // those footnotes to allow 2 way fact <-> footnote navigation.
     for (var id in this.data.facts) {
         var f = new Fact(this, id);
         this._items[id] = f;
         var fns = this.data.facts[id].fn || [];
         fns.forEach((fnid) => {
             var fn = this._items[fnid];
             if (fn === undefined) {
-                fn = new Footnote(this, fnid, "Footnote " + (fncount++).toString());
+                fn = new Footnote(this, fnid, "Footnote " + (fnorder.indexOf(fnid) + 1));
```
Comment (brettkail-wk) [inline iXBRLViewerPlugin/viewer/src/js/report.js]: This appears to be O(n^2) on the number of footnotes.  Not that I expect a large number of footnotes, but it seems like a map<fnid,num> would be nice?

**Comment by derekgengenbacher-wf** [comment]
QA +10 Unit tests pass successfully. Builds successfully. Built locally and tested against docs that have footnotes. Footnotes showed up correctly.

@Workiva/release-management-p

**Pull request merged**